### PR TITLE
fix: array sts server filters

### PR DESCRIPTION
**Pull request by polomani**
basically new tests should be self explanatory
there was one bug, that didn't allow doing any array filtering.
also as a bonus added support for advanced grafana variables. (it allows variable formatting)

### CODE REVIEW

File: src/parser/ts/index.ts [hunk 1]
```diff
@@ -217,24 +217,30 @@ export const getIndicesOfMultiaryFunctionArgs = (route: STSQuery): number[] => {
...
     }
   });
-  return responseArr;
+  return extracted;
+};
+
+export const getServerFilters = (
+  route: STSQuery
+): StringMap[] => {
+  const filter = (_, __, parent) => isServerFilter(parent) || !isSTSFilter(parent);
```
Comment (mrtraser) [inline src/parser/ts/index.ts:236]: Does `isServerFilter(parent)` not enough?

Comment (polomani) [reply src/parser/ts/index.ts:236]: no. it skips `ts{prop=[...arr]}`, as parent just a regular array, not the object

File: src/parser/ts/parser.unit.test.ts [hunk 2]
```diff
@@ -192,7 +202,12 @@ describe('nearley parser', () => {
 
   it('template variables', () => {
     const res = parse(`ts{id=[[asset]]}`);
-    expect(res).toEqual(STS([Filter('id', '$asset')]));
+    expect(res).toEqual(STS([Filter('id', '[[asset]]')]));
+  });
+
+  it('advanced template variable', () => {
+    const res = parse('ts{id=${asset:json}}');
+    expect(res).toEqual(STS([Filter('id', '${asset:json}')]));
```
Comment (mrtraser) [inline src/parser/ts/parser.unit.test.ts:210]: nice!

File: src/parser/ts/index.ts (hunk 1 above)
Comment (mrtraser) [reply src/parser/ts/index.ts:236]: got it

**Pull request merged**